x-pack/test/alerting_api_integration/spaces_only/tests/alerting/find.ts

```diff
@@ -85,5 +95,23 @@ export default function createFindTests({ getService }: FtrProviderContext) {
           data: [],
         });
     });
+
+    it('should filter on string parameters', async () => {
+      await Promise.all([
+        createAlert({ params: { strValue: 'my a' } }),
+        createAlert({ params: { strValue: 'my b' } }),
+        createAlert({ params: { strValue: 'my c' } }),
+      ]);
+
+      const response = await supertest.get(
+        `${getUrlPrefix(
+          Spaces.space1.id
+        )}/api/alerts/_find?filter=alert.attributes.params.strValue:"my b"`
```

Comment:
It's _feeling_ like we're leaking Alerting internal implementation here...

I'm ok with that, as it feels like the overhead of trying to abstract the `attributes` away is probably not worth it.... but I. _would_ like this to have explicit documentation added (asciidocs/dev docs) as I don't think it's intuitive :/

Reply:
@gmmorris Agreed, I always felt like that with the KQL filter. Though, the enforcement of `attributes` is made by the saved objects service so I figured we'd pass it through as we do with other fields.

Under `filter` docs (https://www.elastic.co/guide/en/kibana/master/alerts-api-find.html) the `params` would work the same as any other field but I think it makes sense to document that `params` are analyzed as flattened and point to those docs so users understand the limitations of these fields.

Thoughts?

Reply:
Yeah, makes sense.
Putting myself in the mindset of a developer using our APIs - if the API explains what `params` is and leads me to an API that explains how to query against it - that's fine.

Do we actually explain that the Alerts are stored as SOs? 🤔 
I'm not sure the SO Api makes it clear how to query attributes though 🤔 I'll take a look.

Reply:
From the docs:
>(Optional, string) A KQL string that you filter with an attribute from your saved object. It should look like savedObjectType.attributes.title: “myTitle”. However, If you used a direct attribute of a saved object, such as updatedAt, you will have to define your filter, for example, savedObjectType.updatedAt > 2018-12-22.

I think the only extra mention we need is that they're `flattened` and have filtering limitations.